test/e2e/catalog_e2e_test.go

```diff
@@ -676,7 +677,25 @@ var _ = Describe("Catalog", func() {
 		require.NoError(GinkgoT(), err)
 
 		// Delete the registry pod
-		require.NoError(GinkgoT(), c.KubernetesInterface().CoreV1().Pods(testNamespace).Delete(context.TODO(), name, metav1.DeleteOptions{}))
+		Eventually(func() error {
```

Comment:
While you are editing this test, can you clean up the delete behavior at L649 (to make it retry failures and accept NotFound errors)?

Reply:
In reality we don't even need this anymore since after `It` spec we run the test cleaner